fdb-relational-core/src/test/java/com/apple/foundationdb/relational/recordlayer/RelationalArrayTest.java

```diff
@@ -216,6 +213,116 @@ void testInsertArraysViaQueryPreparedStatement() throws SQLException {
+                assertEquals(1, ps.executeUpdate());
+            }
+
+            try (final var ps = (RelationalPreparedStatement) conn.prepareStatement("SELECT * from T where pk = 2")) {
+                ResultSetAssert.assertThat(ps.executeQuery())
+                        .hasNextRow()
+                        .hasColumn("boolean_null", null)
+                        .hasColumn("integer_null", null)
+                        .hasColumn("bigint_null", null)
+                        .hasColumn("float_null", null)
+                        .hasColumn("double_null", null)
+                        .hasColumn("string_null", null)
+                        .hasColumn("bytes_null", null)
+                        .hasColumn("structure_null", null);
+            }
+        }
+    }
+
+    @Test
+    void testNullColumnDoesNotRequireAValueIfNotGiven() throws SQLException {
+        try (final var conn = DriverManager.getConnection(database.getConnectionUri().toString())) {
+            conn.setSchema(database.getSchemaName());
+            conn.setAutoCommit(true);
+            try (final var ps = ((RelationalPreparedStatement) conn.prepareStatement("INSERT INTO T (pk, boolean_not_null, " +
+                    "integer_not_null, bigint_not_null, float_not_null, double_not_null, string_not_null, bytes_not_null, " +
+                    "structure_not_null, boolean_null, integer_null) VALUES (?pk, ?boolean_not_null, ?integer_not_null, " +
+                    "?bigint_not_null, ?float_not_null, ?double_not_null, ?string_not_null, ?bytes_not_null, ?struct_not_null)"))) {
+                ps.setInt("pk", 2);
+                ps.setArray("boolean_not_null", EmbeddedRelationalArray.newBuilder().addAll(true, false).build());
+                ps.setArray("integer_not_null", EmbeddedRelationalArray.newBuilder().addAll(11, 22).build());
+                ps.setArray("bigint_not_null", EmbeddedRelationalArray.newBuilder().addAll(11L, 22L).build());
+                ps.setArray("float_not_null", EmbeddedRelationalArray.newBuilder().addAll(11.0f, 22.0f).build());
+                ps.setArray("double_not_null", EmbeddedRelationalArray.newBuilder().addAll(11.0, 22.0).build());
+                ps.setArray("string_not_null", EmbeddedRelationalArray.newBuilder().addAll("11", "22").build());
+                ps.setArray("bytes_not_null", EmbeddedRelationalArray.newBuilder().addAll(new byte[]{49}, new byte[]{50}).build());
+                ps.setArray("struct_not_null", EmbeddedRelationalArray.newBuilder().addAll(EmbeddedRelationalStruct.newBuilder().addInt("a", 11).addString("b", "11").build(), EmbeddedRelationalStruct.newBuilder().addInt("a", 22).addString("b", "22").build()).build());
+                assertEquals(1, ps.executeUpdate());
+            }
+        }
+    }
+
+    @Test
+    void testNonNullColumnRequiresValue() throws SQLException {
```

Comment:
This is misleading. The column is not `NOT NULL` since this PR does away with this concept. A column declared in `INSERT INTO T(<columns>)` must be in the `VALUES` clause though.

Reply:
This is fixed as well!